src/Mod/Sketcher/Gui/EditModeConstraintCoinManager.cpp

```diff
@@ -654,25 +655,33 @@ void EditModeConstraintCoinManager::processConstraints(const GeoListFacade & geo
                         if (Constr->SecondPos != Sketcher::PointPos::none) { // point to point distance
                             pnt1 = geolistfacade.getPoint(Constr->First, Constr->FirstPos);
                             pnt2 = geolistfacade.getPoint(Constr->Second, Constr->SecondPos);
-                        } else if (Constr->Second != GeoEnum::GeoUndef) { // point to line distance
+                        } else if (Constr->Second != GeoEnum::GeoUndef) {
                             pnt1 = geolistfacade.getPoint(Constr->First, Constr->FirstPos);
 
                             const Part::Geometry *geo = geolistfacade.getGeometryFromGeoId(Constr->Second);
-                            if (geo->getTypeId() == Part::GeomLineSegment::getClassTypeId()) {
+                            if (geo->getTypeId() == Part::GeomLineSegment::getClassTypeId()) { // point to line distance
                                 const Part::GeomLineSegment *lineSeg = static_cast<const Part::GeomLineSegment *>(geo);
                                 Base::Vector3d l2p1 = lineSeg->getStartPoint();
                                 Base::Vector3d l2p2 = lineSeg->getEndPoint();
                                 // calculate the projection of p1 onto line2
                                 pnt2.ProjectToLine(pnt1-l2p1, l2p2-l2p1);
                                 pnt2 += pnt1;
+
+                            } else if (geo->getTypeId() == Part::GeomCircle::getClassTypeId()) { // circle to circle distance
+                                const Part::Geometry *geo1 = geolistfacade.getGeometryFromGeoId(Constr->First);
+                                auto circleSeg1 = static_cast<const Part::GeomCircle*>(geo1);

```

Comment:
Even if it should be OK, I'm always worrying a bit about static_casting without checking type first. :)

Reply:
You are right.

We need either type checking and static casting, or dynamic casting. I would use type checking and static_casting for consistency (geo is checked, geo1 is not checked).

Reply:
For example:

```
const Part::Geometry *geo1 = geolistfacade.getGeometryFromGeoId(Constr->First);

if(geo1->getTypeId() == Part::GeomCircle::getClassTypeId()) {
    auto circleSeg1 = static_cast<const Part::GeomCircle*>(geo1);
    auto circleSeg2 = static_cast<const Part::GeomCircle*>(geo);
    GetCirclesMinimalDistance(circleSeg1, circleSeg2, &pnt1, &pnt2);
}
```

Reply:
```suggestion
                                if (geo1->getTypeId() == Part::GeomCircle::getClassTypeId()
                                    auto circleSeg1 = static_cast<const Part::GeomCircle*>(geo1);
                                else
                                    break;
```

I anticipate arc distance can be implemented here with a GeomCircle created with the GeomArcOfCircle parameters.

Reply:
You are going to have some issues there. You are going to need a specifically constructed temporary object. Circles are not directly constructable from ArcOfCircles, so you will have to init it (at least via the OCC handle which is not the cleanest).

The code you propose is odd, circleSeg1 is never used. It goes out of scope just when going out of the "if". C++ is a strongly typed language. You cannot "change type". You can fake it to some extent with std::variant, but it has its own caveats too.

In any case, please always use braces for if-else. We didn't use to and lately we have changed this approach, because it leads to many more errors which reviews often miss.

If you want to extend this to arcs (and I guess combinations of arcs and circles), here or in another PR, I would make:

```
template <T1, T2>
bool GetCirclesMinimalDistance(T1 circle_or_arc1, T2 circle_or_arc2, &pnt1, &pnt2);
```

because the API of both is common, you may get away without much adaptation for type.

However, you will still need to make sure to call that function with the right cast-ed types (static o dynamic cast-ed).